python/pyspark/sql/session.py

```diff
@@ -304,8 +306,15 @@ def __init__(
                 and not self._jvm.SparkSession.getDefaultSession().get().sparkContext().isStopped()
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
+                getattr(getattr(self._jvm, "SparkSession$"), "MODULE$").applyModifiableSettings(
```

Comment:
@AngersZhuuuu, this actually shows a lot of new warnings (see also https://github.com/apache/spark/pull/34893). Another reproducer:

```bash
./bin/spark-shell --conf spark.executor.memory=8g --conf spark.driver.memory=8g
```

```python
>>> from pyspark.sql.functions import udf
>>> udf(lambda x: x)("a")
21/12/15 14:03:15 WARN SparkSession: Using an existing SparkSession; the static sql configurations will not take effect.
Column<'<lambda>(a)'>
```

There are too many places like this:

```python
ml/util.py:            self._sparkSession = SparkSession.builder.getOrCreate()
sql/column.py:            spark = SparkSession.builder.getOrCreate()
sql/context.py:            sparkSession = SparkSession.builder.getOrCreate()
sql/readwriter.py:        spark = SparkSession.builder.getOrCreate()
sql/readwriter.py:        spark = SparkSession.builder.getOrCreate()
sql/session.py:                return SparkSession.builder.getOrCreate()
sql/session.py:        return SparkSession.builder.getOrCreate()
sql/streaming.py:        spark = SparkSession.builder.getOrCreate()
sql/streaming.py:        spark = SparkSession.builder.getOrCreate()
sql/udf.py:        spark = SparkSession.builder.getOrCreate()
```

Reply:
Would you mind fixing these please?

If we can't make it by Spark 3.2, I think maybe it's just safer to revert https://github.com/apache/spark/pull/34757 https://github.com/apache/spark/pull/34732 and https://github.com/apache/spark/pull/34559 for now because each patch here will introduce either:
1. Unexpected configuration propagation of static SQL configuration, or
2. Too much warnings

Separately, I still feel https://github.com/apache/spark/commit/8424f552293677717da7411ed43e68e73aa7f0d6 is inefficient. We don't know which configurations don't take affect, or why it keeps complaining (see the example above) for which configuration. We should probably at least print out the keys or lower the level of log.

cc @AngersZhuuuu @yaooqinn @maropu @dongjoon-hyun FYI

Reply:
@xinrong-databricks actually this is more Python side codes. Are you interested in creating a followup?

Reply:
> We should probably at least print out the keys or lower the level of log.

+1

Reply:
Thank you for the head-ups, @HyukjinKwon .

Reply:
Certainly, I will fix it and keep you updated. Thanks!

Reply:
> We should probably at least print out the keys or lower the level of log.

+1

Reply:
Made a PR at https://github.com/apache/spark/pull/35001 👍